python/grass/temporal/aggregation.py

```diff
@@ -359,12 +395,17 @@ def aggregate_by_topology(
             if len(aggregation_list) > 1:
                 # Create the r.series input file
                 filename = gscript.tempfile(True)
-                file = open(filename, "w")
-                for name in aggregation_list:
-                    string = "%s\n" % (name)
-                    file.write(string)
-                file.close()
-
+                with open(filename, "w") as file:
+                    if weighting:
+                        for i, name in enumerate(aggregation_list):
```

Comment:
No need for enumerate + indexing, use zip(aggregation_list, aggregation_weights).

Reply:
```suggestion
                        for name, weight in zip(aggregation_list, aggregation_weights):
```